### PR TITLE
feat: add dataset template

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Add dataset template to scaffold dataset generation script

### CODE REVIEW

File: datasets/new-dataset-using-llm/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Synthetic Data Generation
```
Comment (kimizhu) [inline datasets/new-dataset-using-llm/README.md:1]: Loop Supriya to review it.

Comment (kimizhu) [inline datasets/new-dataset-using-llm/README.md:1]: And better to include the license info as well.

File: datasets/new-dataset-using-llm/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## Generate a synthetic dataset
+**Step 1: Setup the Python Environment**
+
+Install the required Python packages:
+```sh
+pip install -r requirements.txt
+```
+
+**Step 2: Prepare Your Inference Endpoint**
```
Comment (kimizhu) [inline datasets/new-dataset-using-llm/README.md:36]: What is the by default option? should be LLM?

File: datasets/new-dataset-using-llm/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  1. Execute the command palette `AI Toolkit: Download an inference model`.
+  2. Choose a model to generate the dataset, e.g., `Phi-3-mini-4k-cpu-int4-rtn-block-32-acc-level-4-onnx`.
+  3. Set up the environment variables in `generate_qa_dataset.py`:
+      ```python
+      # Define the inference API endpoint, the default is for the local API in AI Toolkit
+      api_endpoint = os.environ.get("API_ENDPOINT", "http://127.0.0.1:5272/v1/chat/completions") 
+      # Define the model to use for the request
+      model = os.environ.get("MODEL", "Phi-3-mini-4k-cpu-int4-rtn-block-32-acc-level-4-onnx")
+      ```
+- **[Option 2]** Use Azure OpenAI Endpoint
```
Comment (kimizhu) [inline datasets/new-dataset-using-llm/README.md:50]: Use the LLM endpoint and Azure OpenAI as an example?

File: datasets/new-dataset-using-llm/prompt/qa.prompt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        "question": "What types of knowledge do LLMs acquire from human language corpora, and what issues do they inherit?",
+        "answer": "LLMs acquire knowledge about syntax, semantics, and ontologies inherent in human language corpora, but they also inherit inaccuracies and biases present in the data they are trained on."
+    }
+]
+
+user:
+{{text}}
+Output with {{num_queries}} QnAs:
+
+
```
Comment (kimizhu) [inline datasets/new-dataset-using-llm/prompt/qa.prompt:47]: extra line?

File: datasets/new-dataset-using-llm/README.md (hunk 1 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md:1]: Add openai and phi license link to the note

File: datasets/new-dataset-using-llm/README.md (hunk 2 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md:43]: Changed the default option to be azure openai

File: datasets/new-dataset-using-llm/README.md (hunk 3 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Changed the default option to be azure openai.

File: datasets/new-dataset-using-llm/prompt/qa.prompt (hunk 4 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/prompt/qa.prompt]: Removed.

File: datasets/new-dataset-using-llm/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+# Synthetic Data Generation
+
+## Overall
+
+Leveraging large language models(LLM) is a crucial technique in synthetic data generation, capable of producing high-quality, diverse, and privacy-compliant datasets. This approach can be applied in various scenarios, including training machine learning models, fine-tuning different language models, and conducting evaluation and testing, among others.
+
+> **Important**: When using a large language model (LLM) to create datasets for developing other models, adherence to the relevant licensing agreements is essential.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:7]: Proposed: When using a large language model (LLM) to create datasets for developing other models, it's important to follow the licensing agreements.

File: datasets/new-dataset-using-llm/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+>
+> This document references Azure OpenAI and Phi-3 as examples. For more details, visit [Azure OpenAI](https://openai.com/policies/business-terms/) and [Phi-3](https://huggingface.co/microsoft/Phi-3-mini-4k-instruct/blob/main/LICENSE).
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:9]: Proposed: This document mentions Azure OpenAI and Phi-3 as examples. For more details, visit [Azure OpenAI]

File: datasets/new-dataset-using-llm/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+## Benefits of synthetic data
+
+- Privacy Protection
+
+  Synthetic data mimics real data without including sensitive information, making it perfect for fields with strict privacy rules.
+
+- Structured and Easy to Use
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:17]: Is this a title?

File: datasets/new-dataset-using-llm/README.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+  You can generate synthetic data on-demand and at a large scale, providing a cost-effective way to get more training data.
+
+- Bias Reduction
+
+  Synthetic data can help reduce bias in AI models by balancing out biased datasets, leading to fairer AI systems.
+
+## Generate a synthetic dataset
+
+**Step 1: Setup the Python Environment**
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:35]: Set up should be two words.

File: datasets/new-dataset-using-llm/README.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+- Structured and Easy to Use
+
+  Synthetic data is often well-structured and can come pre-labeled, simplifying data preparation for machine learning.
+
+- Data Augmentation
+
+  It can fill gaps in sparse datasets, making them richer and more diverse, which is useful for balancing datasets.
+
+- Unlimited Data Generation
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:25]: Unlimited Data Creation

File: datasets/new-dataset-using-llm/README.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+  You can generate synthetic data on-demand and at a large scale, providing a cost-effective way to get more training data.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:27]: You can create synthetic data on-demand and at a large scale, providing a cost-effective way to obtain more training data.

File: datasets/new-dataset-using-llm/README.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+- Bias Reduction
+
+  Synthetic data can help reduce bias in AI models by balancing out biased datasets, leading to fairer AI systems.
+
+## Generate a synthetic dataset
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:33]: ## Create a synthetic dataset

File: datasets/new-dataset-using-llm/README.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+**Step 2: Prepare Your Inference Endpoint**
+
+To set up your language model inference endpoint, you can use Azure OpenAI by default or configure it to use other model.
+
+- **[Option 1]** Use Azure OpenAI Endpoint
+  1. Create an Azure OpenAI resource and deploy a model. Detailed instructions can be found [here](https://learn.microsoft.com/azure/ai-services/openai/how-to/create-resource).
+  2. Configure environment variables
+
+     Set up the necessary environment variables in the `generate_qa_dataset.py` script. This involves specifying the Azure OpenAI inference API endpoint and your API key.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:55]: The script name should be 'create_qa_dataset.py'

File: datasets/new-dataset-using-llm/README.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+      ```python
+      # Define the Azure OpenAI inference API endpoint  
+      api_endpoint = os.environ.get("API_ENDPOINT", "https://<your-resource-name>.openai.azure.com/openai/deployments/gpt-4o/chat/completions?api-version=2024-02-15-preview") 
+
+      # Define the Azure OpenAI API key in env variables
+      api_key = os.environ.get("API_KEY")  
+      ```
+
+- **[Option 2]** Use the Local Model Inference Endpoint in AI Toolkit
+  1. Execute the command palette `AI Toolkit: Download an inference model`.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:66]: Run the command palette `AI Toolkit: Download an inference model`.
Logic: Execute is a forbidden word as per MS style guide.

File: datasets/new-dataset-using-llm/README.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+  2. Choose a model to generate the dataset, e.g., `Phi-3-mini-4k-cpu-int4-rtn-block-32-acc-level-4-onnx`.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:67]: Choose a model to create the dataset, e.g., `Phi-3-mini-4k-cpu-int4-rtn-block-32-acc-level-4-onnx`.

File: datasets/new-dataset-using-llm/README.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+  3. Configure environment variables
+
+     Set up the necessary environment variables in the `generate_qa_dataset.py` script. This involves specifying the local inference API endpoint and the model name.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:70]: The script name should be 'create_qa_dataset.py'

File: datasets/new-dataset-using-llm/README.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+      ```python
+      # Define the inference API endpoint, the default is for the local API in AI Toolkit
+      api_endpoint = os.environ.get("API_ENDPOINT", "http://127.0.0.1:5272/v1/chat/completions") 
+      # Define the model to use for the request
+      model = os.environ.get("MODEL", "Phi-3-mini-4k-cpu-int4-rtn-block-32-acc-level-4-onnx")
+      ```
+
+**Step 3: Update the Context for QA Sets**
+
+Update the context within the script as needed to generate the QA sets.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:81]: Update the context within the script as needed to create the QA sets.

File: datasets/new-dataset-using-llm/README.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+qa_nums = 5
+text = "<your-context>"
+```
+
+**Step 4: Run the script**
+
+Run the script to generate the synthetic dataset:
+
+```sh
+python generate_qa_dataset.py
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:94]: python create_qa_datset.py

File: datasets/new-dataset-using-llm/README.md [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+The generated synthetic dataset will be saved in `qa.jsonl`.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:97]: The created synthetic dataset will be saved in `qa.jsonl`.

File: datasets/new-dataset-using-llm/generate_qa_dataset.py [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+model = os.environ.get("MODEL", None)
+
+# [Option 2] Use the local API in AI Toolkit
+# api_endpoint = os.environ.get("API_ENDPOINT", "http://127.0.0.1:5272/v1/chat/completions") 
+# api_key = os.environ.get("API_KEY", None)
+# model = os.environ.get("MODEL", "Phi-3-mini-4k-cpu-int4-rtn-block-32-acc-level-4-onnx")
+
+# Configuration for QA pair generation
+qa_nums = 3
+text = "Leveraging large language models is a crucial technique in synthetic data generation, capable of producing high-quality, diverse, and privacy-compliant datasets. This approach can be applied in various scenarios, including training machine learning models, fine-tuning different language models, and conducting evaluation and testing, among others."
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/generate_qa_dataset.py:21]: text = " Using large language models(LLM) is a crucial technique in synthetic data generation, capable of producing high-quality, diverse, and privacy-compliant datasets. This technique can be applied in different scenarios, including training machine learning models, fine-tuning different language models, and conducting evaluation and testing."

File: datasets/new-dataset-using-llm/prompt/qa.prompt [hunk 20]
```diff
@@ -0,0 +1,45 @@
+system:
+You are an AI assigned to turn a given text into a QnA set.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/prompt/qa.prompt:2]: You are an AI asked to convert a given text into a QnA set.

File: datasets/new-dataset-using-llm/prompt/qa.prompt [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+Make sure to cover as many details from the given text as you can. 
+Questions should be very specific and detailed.
+Answers should be detailed but no longer than 3 sentences. 
+Provide your answer in JSON structure like this `[{"question": "<The question you generate>", "answer": "<The answer you generate>"}]`.
+Always provide the specified number of QnAs in the output.
+
+user:
+A large language model (LLM) is a computational model capable of language generation or other natural language processing tasks. 
+As language models, LLMs acquire these abilities by learning statistical relationships from vast amounts of text during a self-supervised and semi-supervised training process.
+The largest and most capable LLMs, as of August 2024, are artificial neural networks built with a decoder-only transformer-based architecture, which enables efficient processing and generation of large-scale text data.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/prompt/qa.prompt:13]: As of August 2024, the most advanced LLMs are artificial neural networks built with a decoder-only transformer-based architecture, which enables efficient processing and generation of large-scale text data.

File: datasets/new-dataset-using-llm/prompt/qa.prompt [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+        "question": "What is a large language model (LLM) capable of?",
+        "answer": "A large language model (LLM) is a computational model capable of language generation or other natural language processing tasks."
+    },
+    {
+        "question": "How do LLMs acquire their language processing abilities?",
+        "answer": "LLMs acquire their language processing abilities by learning statistical relationships from vast amounts of text during a self-supervised and semi-supervised training process."
+    },
+    {
+        "question": "What architecture do the largest and most capable LLMs use as of August 2024?",
+        "answer": "The largest and most capable LLMs, as of August 2024, are artificial neural networks built with a decoder-only transformer-based architecture."
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/prompt/qa.prompt:31]: As of August 2024, the most advanced LLMs are artificial neural networks built with a decoder-only transformer-based architecture.

File: datasets/new-dataset-using-llm/README.md (hunk 7 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md:21]: It functions as a concise statement highlighting a particular advantage, similar to a key point.

File: datasets/new-dataset-using-llm/prompt/qa.prompt (hunk 21 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/prompt/qa.prompt:13]: Could we keep the content the same as it appears on [this wikipedia page](https://en.wikipedia.org/wiki/Large_language_model)?

File: datasets/new-dataset-using-llm/README.md (hunk 5 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 6 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 8 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Thanks! Updated these `setup` to be `set up`.

File: datasets/new-dataset-using-llm/README.md (hunk 9 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated `generation` to be `creation`

File: datasets/new-dataset-using-llm/README.md (hunk 10 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 11 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Replaced the `generate` to `create`

File: datasets/new-dataset-using-llm/README.md (hunk 12 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated the script name

File: datasets/new-dataset-using-llm/README.md (hunk 13 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Thanks! Updated all the `execute` to be `run`.

File: datasets/new-dataset-using-llm/README.md (hunk 14 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 15 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 16 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 18 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

File: datasets/new-dataset-using-llm/generate_qa_dataset.py (hunk 19 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/generate_qa_dataset.py]: Updated.

File: datasets/new-dataset-using-llm/prompt/qa.prompt (hunk 20 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/prompt/qa.prompt]: Updated the prompt.

File: datasets/new-dataset-using-llm/prompt/qa.prompt (hunk 22 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/prompt/qa.prompt:31]: Could we keep the content the same as it appears on [this wikipedia page](https://en.wikipedia.org/wiki/Large_language_model)?

Comment (supkasar) [reply datasets/new-dataset-using-llm/prompt/qa.prompt:31]: It's little complex, so I simplified it. But if we've to keep it as per wikipedia, then go for it and if we can change, updating the simplified version would be better.

File: datasets/index.json [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+        {
+            "name": "new-dataset",
+            "diaplayName": "Create a New Dataset",
+            "path": "new-dataset"
+        }
+    ],
+    "scenarioTemplates": [
+        {
+            "name": "new-dataset-using-llm",
+            "diaplayName": "Using LLM to Generate Synthetic Dataset",
```
Comment (supkasar) [inline datasets/index.json:12]: Please change 'generate' to 'create'.

File: datasets/new-dataset-using-llm/README.md [hunk 24]
```diff
@@ -0,0 +1,97 @@
...
+
+```python
+# Configuration for QA pair generation
+qa_nums = 5
+text = "<your-context>"
+```
+
+**Step 4: Run the script**
+
+Run the script to generate the synthetic dataset:
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/README.md:91]: Please change 'generate' to 'create'.

File: datasets/new-dataset-using-llm/prompt/qa.prompt [hunk 25]
```diff
@@ -0,0 +1,45 @@
+system:
+You are an AI asked to convert a given text into a QnA set.
+Both the questions and answers MUST BE directly extracted from the text. 
+Make sure to cover as many details from the given text as you can. 
+Questions should be very specific and detailed.
+Answers should be detailed but no longer than 3 sentences. 
+Provide your answer in JSON structure like this `[{"question": "<The question you generate>", "answer": "<The answer you generate>"}]`.
```
Comment (supkasar) [inline datasets/new-dataset-using-llm/prompt/qa.prompt:7]: Please change 'generate' to 'create'.

File: datasets/index.json (hunk 23 above)
Comment (XiaofuHuang) [reply datasets/index.json]: Updated.

File: datasets/new-dataset-using-llm/README.md (hunk 24 above)
Comment (XiaofuHuang) [reply datasets/new-dataset-using-llm/README.md]: Updated.

**Review by supkasar — APPROVED** [review]

**Pull request merged**